xdsl/dialects/stencil.py

```diff
@@ -359,19 +359,20 @@ def __init__(
             nbounds = bounds
         return super().__init__([nbounds, element_type])
 
-    @classmethod
-    def constr(
-        cls,
-        *,
-        bounds: GenericAttrConstraint[Attribute] | None = None,
-        element_type: GenericAttrConstraint[_FieldTypeElement] | None = None,
-    ) -> (
-        BaseAttr[StencilType[Attribute]]
-        | ParamAttrConstraint[StencilType[_FieldTypeElement]]
-    ):
-        if bounds is None and element_type is None:
-            return BaseAttr(cls)
-        return ParamAttrConstraint(cls, (bounds, element_type))
+
+def StencilTypeConstr(
+    *,
+    bounds: GenericAttrConstraint[Attribute] | None = None,
+    element_type: GenericAttrConstraint[_FieldTypeElement] = AnyAttr(),
+) -> (
+    BaseAttr[StencilType[Attribute]]
+    | ParamAttrConstraint[StencilType[_FieldTypeElement]]
+):
+    if bounds is None and element_type == AnyAttr():
```

Comment:
Not specific to here, but what's the benefit to using `BaseAttr` over a `ParamAttrConstraint`?

Reply:
Perf